srcpkgs/rust/template

```diff
@@ -65,81 +93,92 @@ post_extract() {
 		*-musl) patch -p1 < $FILESDIR/musl.patch ;;
 	esac
 
-	mkdir -p stage0
-	cp -flr ../rustc-*/rustc/* stage0
-	cp -flr ../rust-std-*/rust-std-*/* stage0
-	case "$XBPS_MACHINE" in
-		*-musl) cp -flr ../cargo stage0/bin;;
-		*) cp -flr ../cargo-*/cargo/* stage0;;
-	esac
-
-	# XXX: Cheat Rust build system so we can build rustc using different
-	# version of (prebuilt) stable rustc than preconfigured. It's hack-ish,
-	# but since we're basically rebuilding rustc with the same version,
-	# it's actually safe.
-	# Note: --enable-local-rebuild from #33787 didn't work, don't know why.
-	export LD_LIBRARY_PATH="$wrksrc/stage0/lib"
-	rustc_ver="$($wrksrc/stage0/bin/rustc --version | cut -f2 -d ' ')"
-	rustc_key="$(printf "$rustc_ver" | md5sum | cut -c1-8)"
-	sed -Ei \
-		-e "s/^(rustc):.*/\1: $rustc_ver-1970-01-01/" \
-		-e "s/^(rustc_key):.*/\1: $rustc_key/" \
-		src/stage0.txt
-
-	# Generate config for bootstrap.py to use our prebuilt rustc and cargo
-	# for bootstrapping instead of downloading snapshot from internet.
-	cat > config.toml <<EOF
-[build]
-cargo = "$wrksrc/stage0/bin/cargo"
-rustc = "$wrksrc/stage0/bin/rustc"
-EOF
+	if ! [ "$CROSS_BUILD" ]; then
+		mkdir -p stage0
+		cp -flr ../rustc-*/rustc/* stage0
+		cp -flr ../rust-std-*/rust-std-*/* stage0
+		case "$XBPS_MACHINE" in
+			*-musl) cp -flr ../cargo stage0/bin;;
+			*) cp -flr ../cargo-*/cargo/* stage0;;
+		esac
+	fi
 
 	sed -i /LD_LIBRARY_PATH/d src/bootstrap/bootstrap.py
 }
 
-pre_build() {
-	export CARGO_HOME="$wrksrc/.cargo"
-	export SSL_CERT_FILE=/etc/ssl/certs/ca-certificates.crt
-	export LD_LIBRARY_PATH="$wrksrc/stage0/lib"
-	export PATH="$wrksrc/stage0/bin:$PATH"
-	export MUSL_ROOT=/usr
-	export RUST_BACKTRACE=1
-}
-
 do_configure() {
-	local _triplet
-	case $XBPS_TARGET_MACHINE in
-		i686) _triplet=i686-unknown-linux-gnu;;
-		x86_64) _triplet=x86_64-unknown-linux-gnu;;
-		x86_64-musl) _triplet=x86_64-unknown-linux-musl;;
-		*) _triplet=$XBPS_TARGET_MACHINE;;
-	esac
 	export LD_LIBRARY_PATH="$wrksrc/stage0/lib"
+
 	configure_args="
 		--prefix=/usr
-		--host=${_triplet}
-		--build=${_triplet}
+		--host=${_host_triplet}
+		--target=${_host_triplet}
+		--build=${_build_triplet}
+		--disable-full-bootstrap
 		--release-channel=stable
 		--disable-rpath
 		--disable-docs
 		--disable-codegen-tests
-		--enable-ccache
 		--llvm-root=/usr
-		--enable-local-rust
-		--local-rust-root=$wrksrc/stage0
+		--enable-local-rebuild
```

Comment:
The problem with `--enable-local-rebuild` is that we'll need the a stage0 (or a version installed via xbps) of the same version as the rust version we want to bootstrap. This isn't problematic for everything but x86_64-musl, as we have to compile the stage0 tarball for that ourselves. That means that would have to cross compile rust from x86_64 -> x86_64-musl for the initial stage0 tarball (or compile that manually).

Reply:
Would it be a problem not to set local-rebuild then?

Reply:
Ah, no, that works too, the comment that was previously in the rust template about local-rebuild led me to believe that we needed that.